### DISCUSSION TITLE
Is there any way to compare two objects of different types?

### DISCUSSION THREAD

**Discussion by adoronin** [shouldly/shouldly#806] [Q&A] [answered]
Hi,
I have a few cases when we need to compare two objects of different types property-by-property. The first object is richer than the second. The second object has all the properties of the first object but some of them have different names. We can do this by mapping one to another and then use something like that:

```
FirstType first = new ();
SecondType second = new ();
FirstType secondAsFirstType = // somehow map second to the FirstType, e.g. using AutoMapper 

secondAsFirstType.ShouldBe(first);
```

But maybe is there another nicer way that includes automated mapping? 
Thank you.

**Comment by jnm2** [comment, marked as answer]
No, currently Shouldly does not have a design planned for something like this. The `ShouldBeEquivalentTo` API provides looser defaults than `ShouldBe`, but something like this would need to be opted into with some configuration mechanism. We have to be careful that the default experience wouldn't consider objects of different types to be equal in a surprising way.

**Reply by vladimir-kovalyuk** [reply]
We have to use FluentAssertions's ShouldBeEquivalentOf() for that reason. Nowadays we have more and more difficulties to compare the libraries when starting new project, because Shouldly's pros are being leveled out along the way FluentAssertions is evolving and vice versa. It would be great if the code maintainers of Shouldly improved the check of equivalency. Or maybe elaborated on the extension point that would let a third party library to close this gap.

**Comment by nrjohnstone** [comment]
What about just a new extension method? .ShouldBeEquivalentObjectTo() ? Then it won't change behavior for existing users of ShouldBeEquivalentTo
